packages/actor-http-cache/lib/HttpCacheStorageStream.ts

```diff
@@ -0,0 +1,146 @@
+import { v4 } from 'uuid';
+
+export class HttpCacheStorageStream
+implements IHttpCacheStorage<ReadableStream<Uint8Array>> {
+  private readonly bufferCache: IHttpCacheStorage<Buffer>;
+  private readonly maxBufferSize: number;
+  private readonly mediatorHttpInvalidate?: MediatorHttpInvalidate;
+
+  private incompleteStreams: Record<string, {
+    policy: CachePolicy;
+    init?: ResponseInit;
+    buffers: Uint8Array[];
+    streamId: string;
+    stream: ReadableStream<Uint8Array>;
+  }> = {};
+
+  public constructor(args: IHttpCacheStorageStreamArgs) {
+    this.bufferCache = args.bufferCache;
+    this.maxBufferSize = args.maxBufferSize;
+    this.mediatorHttpInvalidate = args.mediatorHttpInvalidate;
+  }
+
+  public async set(
+    key: Request,
+    value: IHttpCacheStorageValue<ReadableStream<Uint8Array>>,
+    ttl?: number | undefined,
+  ): Promise<void> {
+    if (!value.body) {
+      await this.bufferCache.set(key, { policy: value.policy }, ttl);
+      return;
+    }
+    const streamId = v4();
+    this.incompleteStreams[key.url] = {
+      policy: value.policy,
+      init: value.init,
+      buffers: [],
+      streamId,
+      stream: value.body,
+    };
+    // @ts-expect-error In actuality, this is a NodeJS.Readable
+    value.body.on('data', (chunk: any) => {
+      if (this.incompleteStreams[key.url] && this.incompleteStreams[key.url].streamId === streamId) {
+        this.incompleteStreams[key.url].buffers.push(chunk);
+      }
+    });
+    // @ts-expect-error In actuality, this is a NodeJS.Readable
+    value.body.on('end', async() => {
+      if (!this.incompleteStreams[key.url] || this.incompleteStreams[key.url].streamId !== streamId) {
+        return;
+      }
+      const combinedBuffer = Buffer.concat(this.incompleteStreams[key.url].buffers);
+      if (combinedBuffer.byteLength <= this.maxBufferSize) {
+        await this.bufferCache.set(key, { policy: value.policy, body: combinedBuffer, init: value.init }, ttl);
+      }
+      delete this.incompleteStreams[key.url];
+    });
+  }
+
+  public async get(
+    key: Request,
+  ): Promise<IHttpCacheStorageValue<ReadableStream<Uint8Array>> | undefined> {
+    if (this.incompleteStreams[key.url]) {
+      // Construct a stream from a partial buffer
+      const incompleteStreamInfo = this.incompleteStreams[key.url];
+      let passThrough = new PassThrough();
+      const currentBufferStream = Readable.from(Buffer.concat(incompleteStreamInfo.buffers));
+      passThrough = currentBufferStream.pipe(passThrough, { end: false });
+      // @ts-expect-error In actuality, this is a NodeJS.Readable
+      passThrough = incompleteStreamInfo.stream.pipe(passThrough, { end: false });
+      // @ts-expect-error In actuality, this is a NodeJS.Readable
+      incompleteStreamInfo.stream.on('end', () => passThrough.emit('end'));
+      // @ts-expect-error In actuality, this is a NodeJS.Readable
+      ActorHttp.normalizeResponseBody(passThrough);
```

Comment:
Instead of calling this, does `ActorHttp.toWebReadableStream` already do what we want?

If it does, then we probably don't need to create this `normalizeResponseBody` function anymore.

Reply:
`ActorHttp.toWebReadableStream` cannot be used because of the problems listed below.